### ISSUE TITLE
Review the need to adjust final selection in Expr::applyFunctionWithPeeling

### ISSUE THREAD

**Issue by mbasmanova** [facebookincubator/velox#2595]
See https://github.com/facebookincubator/velox/pull/2592#pullrequestreview-1114176395

**Comment by bikramSingh91** [comment]
@mbasmanova please correct me if I am wrong.
I looked into this yesterday, and was trying to come up with a scenario where not adjusting final selection in applyFunctionWithPeeling() would result in an issue. Turns out the actual value of the final selection does not matter at this point. What matters is whether it is set or not. This is because the final selection rowset is used to materialize lazy vectors, however when we get to this point, the input vectors would have been loaded and the applying function only needs to check whether finalSelection is set or not so that it can preserve partially populated results.

TLDR: we dont need to update final selection since its actual value is not used.
However, we would benefit from adding a comment here that explains why it is not required since this function is virtually identical to peelEncodings() and a similar question can come up in the future (the big difference here is that peelEncodings() only applied to input columns whereas this applies to intermediate results)

**Comment by mbasmanova** [comment]
@bikramSingh91 Bikram, thank you for looking into this and writing up a detailed summary of your findings. This makes sense to me and I agree that adding a comment would be beneficial.
